File: 1002 - Find Common Characters.py
class Solution:
    def commonChars(self, words: list[str]) -> list[str]:
        allChar = set()
        count = dict()
        for c in words[0]:
            count[c] = count.setdefault(c, 0) + 1
        for word in words:
            for c in count.keys():
                count[c] = min(count[c], word.count(c))
        res = []
        for c in count.keys():
            for i in range(count[c]):
                res.append(c)
        return res

File: 1002 - Find Common Characters.java
import java.io.CharArrayReader;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

class Solution {
    public List<String> commonChars(String[] words) {
        Map<Character, Integer> count = new HashMap<>();
        for (char c : words[0].toCharArray()) {
            count.put(c, count.getOrDefault(c, 0) + 1);
        }
        for (int i = 1; i < words.length; i++) {
            for (char c : count.keySet()) {
                int cnt = count(words[i], c);
                count.put(c, Math.min(cnt, count.get(c)));
            }
        }
        List<String> res = new ArrayList<>();
        for (Character c : count.keySet()) {
            for (int i = 0; i < count.get(c); i++) {
                res.add(Character.toString(c));
            }
        }
        return res;
    }
    public int count(String word, char c) {
        int res = 0;
        for (char a : word.toCharArray()) {
            if (a == c) res++;
        }
        return res;
    }
}